viadot/sources/mindful.py

```diff
@@ -0,0 +1,251 @@
+            end_date (datetime, optional): End date of the resquest. Defaults to None.
+            date_interval (int, optional): How many days are included in the request.
+                If end_date is passed as an argument, date_interval will be invalidated. Defaults to 1.
+            limit (int, optional): The number of matching interactions to return. Defaults to 1000.
+
+        Raises:
+            ValueError: Not available endpoint.
+            APIError: Failed to download data from endpoint.
+
+        Returns:
+            Response: request object with the response from the Mindful API.
+        """
+        if endpoint not in self.ENDPOINTS:
+            raise ValueError(
+                f"endpoint {endpoint} not available through viadot connection, yet."
+            )
+
+        params = {
+            "_limit": limit,
+            "start_date": f"{start_date}",
+            "end_date": f"{end_date}",
+        }
+
+        if endpoint == "surveys":
+            del params["start_date"]
+            del params["end_date"]
+        else:
+            start_date, end_date = self._check_timestamp(
+                start_date, end_date, date_interval
+            )
+            params["start_date"] = start_date
+            params["end_date"] = end_date
+
+        response = self._mindful_api_response(
+            endpoint=endpoint,
+            params=params,
+        )
+
+        if response.status_code == 200:
+            self.logger.info(
+                f"Succesfully downloaded {endpoint} data from mindful API."
+            )
+            data = StringIO(response.content.decode("utf-8"))
+        elif response.status_code == 204 and not response.content.decode():
+            self.logger.warning(
+                f"Thera are not {endpoint} data to download from {start_date or ' --- '} to {end_date  or ' --- '}."
+            )
+            data = json.dumps({})
+        else:
+            self.logger.error(
+                f"Failed to downloaded {endpoint} data. - {response.content}"
+            )
+            raise APIError(f"Failed to downloaded {endpoint} data.")
+
+        data_frame = pd.read_json(data)
+        if data_frame.empty:
+            self._handle_if_empty(
+                if_empty="warn",
+                message=f"No data was got from the endpoint: {endpoint}, days from {start_date} to {end_date}",
+            )
+
+        return data_frame
+
+    def to_file(
+        self,
+        data_frame: pd.DataFrame,
+        file_path: str = "",
+        file_name: Optional[str] = None,
+        file_extension: Literal["parquet", "csv"] = "csv",
+        sep: str = "\t",
+    ) -> str:
+        """Save Mindful response data to file and return filename.
+
+        Args:
+            data_frame (pd.DataFrame):
+            file_path (str, optional): Path where to save the file locally. Defaults to ''.
+            file_name (str, optional): Name of the file without extension. Defaults to None.
```

Comment:
Looking at these two together, does that mean `file_path` is the path but without the last part (file name), which has to be specified in this other param? This is unclear from these params and their docstrings. Would be useful to also add an example usage of this method in the `Examples` section of the docstring, then it would be clear.

Reply:
added.